diesel_tests/tests/serialize_as.rs

```diff
@@ -0,0 +1,65 @@
+use crate::schema::*;
+use diesel::*;
+use diesel::serialize::{Output, ToSql};
+use std::io::Write;
+use diesel::backend::Backend;
+
+#[derive(Debug, FromSqlRow, AsExpression)]
+#[sql_type = "sql_types::Text"]
+struct UppercaseString(pub String);
+
+impl From<String> for UppercaseString {
+    fn from(s: String) -> Self {
+        UppercaseString(s.to_uppercase())
+    }
+}
+
+impl From<&String> for UppercaseString {
+    fn from(s: &String) -> Self {
+        UppercaseString(s.to_uppercase())
+    }
+}
```

Comment:
Why do we need both impls?

Reply:
I'd love to not have the 2nd one because I don't think it makes sense but it didn't compile without it :(